### ISSUE TITLE
No backup images found in image directory

### ISSUE THREAD

**Issue by bloatware99** [4nubhav/WhatsDeleted#6]
I enabled the setting , allow notifucation access

**Comment by 4nubhav** [comment]
Notification access is for messages.
For images, make sure:

- You've granted the storage permission.
- Media Observer switch is on. (If on, a persistent notification is also displayed)

Also note:

- Images will be saved to the backup directory **only if they were downloaded.**

**Comment by bloatware99** [comment]
I already did the settings kindly refer attached screenshots , I downloaded the images got deleted by sender even also I did not backup bro.-- 
 Securely sent with Tutanota. Get your own encrypted, ad-free mailbox: 
 https://tutanota.com


Jul 10, 2020, 12:16 by notifications@github.com:

**Comment by 4nubhav** [comment]
I haven't received any screenshots, can you send again?

**Comment by bloatware99** [comment]
I resended did you recieved screenshot now.
-- 
 Securely sent with Tutanota. Get your own encrypted, ad-free mailbox: 
 https://tutanota.com


Jul 10, 2020, 22:03 by notifications@github.com:

**Comment by 4nubhav** [comment]
Nope.
Can you comment directly in GitHub and attach your screenshots?

**Comment by bloatware99** [comment]
Yes

**Comment by bloatware99** [comment]
@4nubhav ![Screenshot_20200710-183929.png](https://user-images.githubusercontent.com/65462601/87183789-6d582480-c304-11ea-8b92-a6b64c889d96.png)
![Screenshot_20200710-183950.png](https://user-images.githubusercontent.com/65462601/87183803-75b05f80-c304-11ea-8f89-0700b125786d.png)

**Comment by 4nubhav** [comment]
Can you tell me which phone & OS you're using?
Are the messages getting backed up?
Also, are you opening the image directory from your gallery app or file manager app?

**Comment by bloatware99** [comment]
I'm using motorola one power , android 10 OS.
Still I didn't get backup.
I'm opening image directory in go files app by google 



Jul 11, 2020, 09:30 by notifications@github.com:

**Comment by 4nubhav** [comment]
Okay.
Can you please click on the `View` button beside `Message log` and let me know if you see any messages or not?

**Comment by bloatware99** [comment]
> Okay.
> Can you please click on the `View` button beside `Message log` and let me know if you see any messages or not?

![obscura1594442248617.jpg](https://user-images.githubusercontent.com/65462601/87216551-6fe76800-c35e-11ea-8f75-84bc213c4d62.jpg)

**Comment by 4nubhav** [comment]
So the messages are getting backed up but the images aren't. I'll look into it & let you know.

**Comment by bloatware99** [comment]
Thanks in advance bro.

**Comment by 4nubhav** [comment]
What is the folder path where WhatsApp images are downloaded on your phone?
Is it `Internal storage/WhatsApp/Media/WhatsApp Images/` ?

**Comment by bloatware99** [comment]
Internal/whatsApp/Media/WhatsApp Images

But Into the whatsApp images folder there are Private and Sent folders

Jul 12, 2020, 20:20 by notifications@github.com:

**Comment by 4nubhav** [comment]
Hi! Will you be able to get logs?

**Comment by bloatware99** [comment]
Yes bro I get logs But WhatsDeleted Images folder are empty.

**Comment by 4nubhav** [comment]
I'm talking about error logs actually.
The app logs a message to `LogCat` every time it fails to save an image to the `WhatsDeleted Images` folder.
Those logs can be obtained through `adb logcat`. I need them to fix whatever the error that's happening.
I can assist if you need any help in getting those logs.

**Comment by bloatware99** [comment]
> I'm talking about error logs actually.
> The app logs a message to `LogCat` every time it fails to save an image to the `WhatsDeleted Images` folder.
> Those logs can be obtained through `adb logcat`. I need them to fix whatever the error that's happening.
> I can assist if you need any help in getting those logs.

https://del.dog/uceecuhimo.txt

**Comment by bloatware99** [comment]
I attached the logcat bro

**Comment by 4nubhav** [comment]
Did you get the logs through `adb`?

**Comment by felipe19930** [comment]
Hi @4nubhav, I'm having the same issue on my Moto E7 phone. WhatsDeleted is able able to get message logs, but when people send me images, they're not being saved to the 'Internal Storage/WhatsDeleted Images' folder.

And yes, the Media Observer service/notification is showing up and I do have auto-download enable for images in WhatsApp.

Here's the adb log, I started capturing it right before someone one sent me an image on WhatsApp, and I waited a few seconds for the image to appear in the notifications so I could stop logcat:

[logcat.txt](https://gist.github.com/felipe19930/b3c96d46308815e8447271bfdbefbddd/raw/31243addd316032936f60ad7dddd8a8afc81f3f5/logcat.txt)

Though I don't see anything useful there.
